app/utilities/metadata_parser.py

```diff
@@ -101,8 +102,9 @@ def validate_schema_name(self, data, **kwargs):
                 )
 
     @post_load
-    def update_schema_name(self, data, **kwargs):
-        # pylint: disable=no-self-use, unused-argument
+    def update_schema_name(  # pylint: disable=no-self-use, unused-argument
+        self, data: dict, **kwargs: Any
```

Comment:
`dict` > `MutableMapping`. `dict` is more concrete than `MutableMapping`, as per our style guide, args should be abstract.